src/Datadog.Trace.ClrProfiler.Managed/Instrumentation.cs

```diff
@@ -126,5 +130,37 @@ private static void StartDiagnosticManager()
             DiagnosticManager.Instance = diagnosticManager;
         }
 #endif
+
+        private static void TryLoadVendorPlugin()
+        {
+            string pluginName = GlobalSettings.Source.VendorPluginName;
+            if (!string.IsNullOrWhiteSpace(pluginName))
+            {
+                string pluginPath = Path.Combine(
+                    Path.GetDirectoryName(typeof(Instrumentation).Assembly.Location),
+                    pluginName);
```

Comment:
I feel that specifying full path more flexible and can be safer at the same time.

I think it should locate files in the same way as `OTEL_DOTNET_TRACER_HOME` and `OTEL_INTEGRATIONS`.

Reply:
You can't specify full path, it should load the correct version (runtime version) of the plugin.

Reply:
Runtime folder logic is stored in Loader assembly. Not sure we want to redesign that heavily to reuse the same logic.

Reply:
> Runtime folder logic is stored in Loader assembly. Not sure we want to redesign that heavily to reuse the same logic.

I think it would be good, but no worth starting with it.

So basically I think we initially support only the env var value to be equal to `OTEL_DOTNET_TRACER_HOME` value. And then we can reimplment the assembly loader so that it can be used for plugin loading. Does it make any sense?

Reply:
I want to say that this should work upfront.
```
OTEL_DOTNET_TRACER_HOME=/home/me/tracer
OTEL_DOTNET_TRACER_PLUGINS=/home/me/tracer

/home/me/tracer
|
- integrations.json
- ...Native.dll
- netcoreapp3.1
|--- ...Tracer.dll
|--- ...plugin.dll 
```

And this should NOT work initially, but after could work after we reuse the assembly loader.

```
OTEL_DOTNET_TRACER_HOME=/home/me/tracer
OTEL_DOTNET_TRACER_PLUGINS=/home/plugins

/home/me/tracer
|
- integrations.json
- ...Native.dll
- netcoreapp3.1
|--- ...Tracer.dll 

/home/plugins
|
- netcoreapp3.1
|--- ...plugin.dll 
```

Reply:
I like the idea of have a separate plugin location. That would make it easier for people that have their own custom plugins (for whatever reason) outside of a vendor to be able to update their installation and potentially not remove their current plugins.

Reply:
Should we divert back to the original idea? Instead of specifying plugin files, we'll just specify plugin dir. We can use the same logic to determine the runtime folder without bigger changes. 
Any concerns about security? - it will autoload all dlls in plugin dir.

Reply:
> Any concerns about security? - it will autoload all dlls in plugin dir.

At a minimum, the mechanism has to be documented. Because it can always be used as an attack vector. No matter how it can be secured, it always allows ACE (Arbitraty Code Execution).

If we are not happy with autoloading all dlls in a plugin directory, we can introduce more sophisticated `plugins.json` (like `integrations.json`) where we could allow specifying plugins in a more strict manner. We could even introduce stuff like plugin checksums to make sure that nothing has modified the plugins and that they are exactly what we expect. I cannot imagine a better (yet still being quite simple) defensive mechanism.

Reply:
Json config sounds better, regarding that we can have different security options there. 
Then there is an option to use also full paths to plugins - there could be a use case where plugin is missing for certain framework versions. 

```
{
   ...
   "net45": [
      path/to/my/plugin1.dll,
   ],
   ...
   "netcoreapp3.1": [
      path/to/my/plugin1.dll,
      { path: path/to/my/plugin2.dll, checksum: <check_sum> }
   ]
}
```